src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -433,6 +438,928 @@ unsafe struct S
+    {
+        nop
+        ldftn void Program::Called()
+        ret
+    } // end of method Program::Main
+
+    .method private hidebysig static 
+        void Called () cil managed 
+    {
+        nop
+        ldstr ""Called""
+        call void [mscorlib]System.Console::WriteLine(string)
+        nop
+        ret
+    } // end of Program::Called
+
+    .method public hidebysig specialname rtspecialname
+        instance void .ctor() cil managed
+    {
+            ldarg.0
+            call instance void[mscorlib] System.Object::.ctor()
+            nop
+            ret
+    } // end of Program::.ctor
+}
+";
+
+            var source = @"
+unsafe class Caller
+{
+    static delegate*<void> _field;
+
+    public unsafe static void Main()
+    {
+        _field = Program.LoadPtr();
+        Call();
+    }
+
+    public unsafe static void Call()
+    {
+        _field();
+    }
+}";
+
+            var verifier = CompileAndVerifyFunctionPointersWithIl(source, ilStub, expectedOutput: "Called");
+            verifier.VerifyIL("Caller.Call()", @"
+{
+  // Code size       11 (0xb)
+  .maxstack  1
+  IL_0000:  ldsfld     ""delegate*<void> Caller._field""
+  IL_0005:  calli      0x3
+  IL_000a:  ret
+}");
+        }
+
+        [Fact]
+        public void CalliOnFieldArgs()
+        {
+            var ilStub = @"
+.class public auto ansi beforefieldinit Program
+    extends [mscorlib]System.Object
+{
+    // Methods
+    .method public hidebysig static 
+        method void *(string) LoadPtr () cil managed 
+    {
+        nop
+        ldftn void Program::Called(string)
+        ret
+    } // end of method Program::Main
+
+    .method private hidebysig static 
+        void Called (string arg) cil managed 
+    {
+        nop
+        ldarg.0
+        call void [mscorlib]System.Console::WriteLine(string)
+        nop
+        ret
+    } // end of Program::Called
+
+    .method public hidebysig specialname rtspecialname
+        instance void .ctor() cil managed
+    {
+            ldarg.0
+            call instance void[mscorlib] System.Object::.ctor()
+            nop
+            ret
+    } // end of Program::.ctor
+}
+";
+
+            var source = @"
+unsafe class Caller
+{
+    static delegate*<string, void> _field;
+
+    public unsafe static void Main()
+    {
+        _field = Program.LoadPtr();
+        Call();
+    }
+
+    public unsafe static void Call()
+    {
+        _field(""Called"");
+    }
+}";
+
+            var verifier = CompileAndVerifyFunctionPointersWithIl(source, ilStub, expectedOutput: "Called");
+            verifier.VerifyIL("Caller.Call()", @"
+{
+  // Code size       18 (0x12)
+  .maxstack  2
+  .locals init (delegate*<string,void> V_0)
+  IL_0000:  ldsfld     ""delegate*<string,void> Caller._field""
+  IL_0005:  stloc.0
+  IL_0006:  ldstr      ""Called""
+  IL_000b:  ldloc.0
+  IL_000c:  calli      0x4
+  IL_0011:  ret
+}");
+        }
+
+        [Theory(Skip = "PROTOTYPE(func-ptr)")]
```

Comment:
I was having some issues figuring out the required il for this test and wanted to get the PR out before I left, so I skipped it for now.